Throw an exeception instead of simply exiting when there are fatal errors.

 -- Rafael Laboissiere <rafael@localhost>  Tue, 18 Nov 2008 08:34:16 +0100

--- a/include/ANN/ANN.h
+++ b/include/ANN/ANN.h
@@ -830,4 +830,9 @@
 
 DLL_API void annClose();		// called to end use of ANN
 
+struct ANN_exception {
+  std::string msg;
+  ANN_exception(const std::string& _msg) : msg(_msg) {}
+};
+
 #endif
--- a/src/ANN.cpp
+++ b/src/ANN.cpp
@@ -168,8 +168,8 @@
 void annError(char *msg, ANNerr level)
 {
 	if (level == ANNabort) {
-		cerr << "ANN: ERROR------->" << msg << "<-------------ERROR\n";
-		exit(1);
+         //            cerr << "ANN: ERROR------->" << msg << "<-------------ERROR\n";
+               throw ANN_exception(msg);
 	}
 	else {
 		cerr << "ANN: WARNING----->" << msg << "<-------------WARNING\n";
